draft-frindell-moq-chat.md

```diff
@@ -55,115 +55,81 @@ receive messages published by others.
+* user-id - the user ID
+* device-id - a unique identifier for each device for the user.  This allows
+              the same user to join the chat from multiple devices
+* timestamp - the timestamp in seconds when the track started.  This allows a
+              stateless client to start publishing without accidentally
+              overwriting a previously sent group and object.
 
-To join the chat a participant sends a SUBSCRIBE message to moq-chat/id in order
-to get the participant list and subsequent participant updates.  This SUBSCRIBE
-requires the AUTHORIZATION_INFO to contain the participant's ASCII string
-username.  It is RECOMMENDED the participant request the catalog starting from
-the beginning of the current Group. The moq-chat server will reply with an
-SUBSCRIBE_OK if joining was successful or SUBSCRIBE_ERROR if the join failed.
+## Joining the Chat
 
-The participant also sends an ANNOUNCE message to the server with a track
-namespace of moq-chat/id/participant/username, and an empty Track Name.
+To join the chat a participant sends a SUBSCRIBE_NAMESPACE message to the relay
+with a namesapce prefix ("moq-chat", \<id\>) in order to receive all ANNOUNCE
+message from each participant in the chat and UNANNOUNCE messages when
+participants leave.
 
-When a moq-chat server receives a SUBSCRIBE and sends an SUBSCRIBE OK, it MUST
-update the catalog.  It can either publish a new Group with the updated
-particiant list, a delta encoding against the current Group, or both.
+The participant also sends an ANNOUNCE message for their chat track namespace.
 
 ## Subscribing to Chat Messages
 
-After receiving the most recent catalog information, a client SHOULD subscribe
-to the track for each active participant.
-
-Upon receiving a delta update removing a participant, a client SHOULD
-unsubscribe from that track if it had previously subscribed.
+When receiving an ANNOUNCE that matches the chat prefix, the client extracts the
+client's user-id, device-id and timestamp from the third, fourth and fifth tuple
+elements, respectively.  The client SHOULD subscribe to the latest timestamp
+track for each (user-id, device-id) pair.
 
-Upone receiving a delta update adding a participant, a client should subscribe
-to the new track.
+Upon receiving an UNANNOUNCE, a client SHOULD UNSUBSCRIBE from that
+matching track if it had previously subscribed.
 
 ## Chat Messages
 
 Each chat message is sent in a new Group and new Object.  The format of a chat
-chat message for version 1 catalogs is UTF-8 Encoded text.  There is no limit to
-the length of a chat message beyond those imposed on QUIC streams.
+chat message this draft is UTF-8 Encoded text.  There is no limit to
+the length of a chat message beyond those imposed on QUIC streams.  Chat clients
+MUST send an END_OF_GROUP message for each Group.
+
+The starting Group ID for each track starts at 0 and increments by 1.  The
+Object ID for each chat message starts at 0 and increments by 1.
```

Comment:
curious to hear your thinking on how does search for set of chats ( say chat with alice last week) work (when cached)

Reply:
In this design, track names contain a timestamp of when the chat started.  So you could request all the chat tracks that cover that range, plus the last one that started before.

Christian rightly pointed out that this protocol should have a timestamp in the object payload also which would help support that use case. 

@suhasHere Do you have an alternate proposal?